src/main/java/uk/gov/companieshouse/company/profile/service/CompanyProfileService.java

```diff
@@ -332,4 +332,46 @@ private void setUpdateIfNotNullOtherwiseRemove(Update update, String key, Object
             update.unset(key);
         }
     }
+
+    /**
+     * Check if psc statements link exists already on document and
+     * call addLink if this is false.
+     *
+     * @param request Data required to identify type of link
+     */
+    public void addPscStatementsLink(LinkRequest request) {
+        if (!StringUtils.isBlank(getDocument(request.getCompanyNumber())
+                .getCompanyProfile()
+                .getLinks()
+                .getPersonsWithSignificantControlStatements())) {
+            logger.error("PSC statements link for company profile already exists");
+            throw new ResourceStateConflictException("Resource state conflict; "
+                    + "PSC statements link already exists");
+        } else {
+            addLink(request);
```

Comment:
Maybe not one for this story but it feels like there's cleaner ways to handle this than adding a new method every time for add and delete links just for a null check

Reply:
Maybe we could look at passing in the method to check on as part of the link request

Reply:
Tech debt ticket created for this https://companieshouse.atlassian.net/browse/DSND-1615